### PR TITLE
fix: rename swap

### LINKED ISSUE #3263: Rename `swap-endpoint` to better name
`swap-endpoint` represent Bee's blockchain connectivity and nowadays is not linked anymore only to SWAP protocol, but also postage-stamps, in close future to the storage loterry etc. It is quite confusing naming, so I would suggest renaming it to something like `rpc-endpoint`. It could be done with deprecation, eq. supporting for some time (until another breaking release?) both options with priority over the new name, communicating this to users and eventually removing `swap-endpoint`.

### PR DESCRIPTION
**Pull request by umerm-work**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [ ] My change requires updating the Open API specification and/or changing its version, and I've done it.
- [ ] I have added tests to cover my changes.
- [ ] I have filled out the description and linked the related issues.

### Description
Rename swap-endpoint to blockchain-rpc-endpoint.Closes  #3263
### Screenshots (if appropriate):

- - -
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/3345)

### CODE REVIEW

**Review by notanatol — APPROVED** [review]

File: pkg/node/node.go [hunk 1]
```diff
@@ -149,7 +149,8 @@ type Options struct {
 	ResolverConnectionCfgs     []multiresolver.ConnectionConfig
 	RetrievalCaching           bool
 	BootnodeMode               bool
-	SwapEndpoint               string
+	SwapEndpoint               string // deprecated: use rpc endpoint instead
+	RpcEndpoint                string
```
Comment (vladopajic) [inline pkg/node/node.go:153]: Could `SwapEndpoint` be removed from `Options` as well?

If we are deprecating this option we should stop using it in all packages other then `cmd` were we need to keep it for backward compatibility.

**Review by mrekucci — COMMENTED** [review]
Not a good idea, especially renaming it to `rpc-endpoint` as it has a broad meaning. The prefix `swap` is narrower, though I agree it's not the best name either. It would be better to come up with a more specific and suitable name.

**Comment by mfw78** [comment]
There may additionally be the time where the resolver-options​ actually specifies an rpc-endpoint​, and this would be an ethereum mainnet​ endpoint, as opposed to Gnosis Chain.

Perhaps something like settlement-endpoint​, or incentives-endpoint​ to highlight the use of said endpoint?

------- Original Message -------

File: pkg/node/node.go (hunk 1 above)
Comment (umerm-work) [reply pkg/node/node.go]: Yes, we can.

**Review by vladopajic — COMMENTED** [review]
LGTM; apart from different opinions on option name

**Comment by umerm-work** [comment]
Now the endpoint name will be `blockchain-rpc-endpoint` as finalized in #3263

**Review by mrekucci — CHANGES_REQUESTED** [review]
This PR contains 106 files, most of which are unrelated. Also, some checks are failing.

**Comment by umerm-work** [comment]
> This PR contains 106 files, most of which are unrelated. Also, some checks are failing.

Done

**Review by mrekucci — COMMENTED** [review]
There are other places such as homebrew, packaging, etc. where this needs to be reflected. Look at other PRs where a parameter has been added or removed to see all the places.

**Review by umerm-work — COMMENTED** [review]
Done

> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/3345#-:-NEqEBRM4CMMxht6Al1G:bb6un1j)* status: 0 of 8 files reviewed, 1 unresolved discussion (waiting on @aloknerurkar and @vladopajic)

File: packaging/bee.yaml [hunk 2]
```diff
@@ -60,8 +60,10 @@ password-file: /var/lib/bee/password
 # resolver-options: []
 ## enable swap (default true)
 # swap-enable: true
-## swap ethereum blockchain endpoint (default "")
+## swap ethereum blockchain endpoint (default "") [deprecated]
```
Comment (mrekucci) [inline packaging/bee.yaml:63]: I would also remove "ethereum". The same is in other places.

Comment (umerm-work) [reply packaging/bee.yaml]: Done

**Review by mrekucci — APPROVED** [review]

**Comment by AuHau** [comment]
@umerm-work can you please also update this in Docs?
https://github.com/ethersphere/bee-docs/blob/master/docs/working-with-bee/configuration.md#--swap-endpoint
(there might be other places where this is mentioned as well)

**Comment by umerm-work** [comment]
> @umerm-work can you please also update this in Docs? https://github.com/ethersphere/bee-docs/blob/master/docs/working-with-bee/configuration.md#--swap-endpoint (there might be other places where this is mentioned as well)

Sure will update it as well.

**Comment by AuHau** [comment]
Thanks, and actually now looking at the code, could there be also a warning in logs that the user is using the old and deprecated flag?

**Pull request merged**